AtReconstruction/AtCopyAuxTreeTask.cxx

```diff
@@ -0,0 +1,89 @@
+#include "AtCopyAuxTreeTask.h"
+
+#include "AtRawEvent.h"
+
+#include <FairLogger.h>
+#include <FairRootFileSink.h>
+#include <FairRootManager.h>
+#include <FairRunAna.h>
+#include <FairSink.h> // for FairSink
+#include <FairTask.h>
+
+#include <TChain.h>
+#include <TClonesArray.h>
+#include <TObject.h>
+
+bool AtCopyAuxTreeTask::SetInputTree(TString fileName, TString treeName)
+{
+   if (fInputTree != nullptr) {
+      LOG(error) << "Input tree was already set! Add more trees using the "
+                 << "function AddInputTree.";
+      return false;
+   }
+   // Add the tree to the input
+   fInputTree = std::make_unique<TChain>(treeName);
+
+   // If there is a tree with the correct name in the file
+   return AddInputFile(fileName);
+}
+
+bool AtCopyAuxTreeTask::AddInputFile(TString fileName)
+{
+   if (fInputTree == nullptr) {
+      LOG(fatal) << "Input TChain was not initialized using the "
+                 << "function SetInputTree.";
+      return false;
+   }
+
+   return (fInputTree->Add(fileName, -1) == 1);
+}
+
+InitStatus AtCopyAuxTreeTask::Init()
+{
+   // Register the input and output branches with the IO manager
+   FairRootManager *ioMan = FairRootManager::Instance();
+   if (ioMan == nullptr) {
+      LOG(fatal) << "Cannot find RootManager!";
+      return kFATAL;
+   }
+
+   fRawEventArray = dynamic_cast<TClonesArray *>(ioMan->GetObject(fRawEventBranchName));
+   if (fRawEventArray == nullptr) {
+      LOG(fatal) << "Cannot find AtRawEvent array in branch " << fRawEventBranchName << "!";
+      return kFATAL;
+   }
```

Comment:
Why is there here? Is there a reason to require a raw event is in the tree?

Reply:
This should be an AtBaseEvent, not and AtRawEvent. No reason to require the user to have a raw event in the ttree when all we are doing is looking at if an event is good or not.

Reply:
This has been changed to an AtBaseEvent, now (didn't realize AtBaseEvent had the IsGood checking in it. This is definitely a good change)